lib/backend/k8s/conversion/conversion.go

```diff
@@ -212,6 +213,37 @@ func (c Converter) PodToWorkloadEndpoint(pod *kapiv1.Pod) (*model.KVPair, error)
 	labels[apiv3.LabelNamespace] = pod.Namespace
 	labels[apiv3.LabelOrchestrator] = apiv3.OrchestratorKubernetes
 
+	// Pull out floating IP annotation
+	var floatingIPs []apiv3.IPNAT
+	if annotation, ok := pod.Annotations["cni.projectcalico.org/floatingIPs"]; ok {
+
+		// Parse Annotation data
+		var ips []string
+		err := json.Unmarshal([]byte(annotation), &ips)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse '%s' as JSON: %s", annotation, err)
+		}
+
+		// Get target for NAT
+		podip, podnet, err := cnet.ParseCIDROrIP(ipNets[0])
```

Comment:
We already do some of this validation when we figure out the IP nets above: https://github.com/bradbeam/libcalico-go/blob/e8f5a298c93e95facc37ea2081ee9a1320933b19/lib/backend/k8s/conversion/conversion.go#L150:20

The one bit that function doesn't do is check the netmask. I'd suggest either dropping the netmask check since the data from the k8s API should be valid anyway, or moving the netmask check into `GetPodIPs`

Reply:
Something to consider - what will need to change here when multiple pod IPs are supported?

Reply:
Not sure I have an answer for that. I'd guess it really depends on the schema for multiple interfaces, do you have any insight into what that might look like?

Reply:
re: netmask check, it was added from the earlier review.  @fasaxc any concerns about dropping it given what casey mentioned?

Reply:
woops, sorry for conflicting advice! I'll defer to @fasaxc's review.

Reply:
> do you think that's a good-enough answer? Are there any other similar annotations we could take inspiration from?

I'm sure we can find some way of extending it in the future, though I don't yet know what multiple IPs might look like. There may be bits of metadata attached to them we could use. Worried that requiring knowing what the internal IP will be is not a very friendly UX since that is typically assigned to a pod at random.